src/FluentResults/Results/Result.cs

```diff
@@ -148,6 +148,56 @@ public Result<TNewValue> ToResult<TNewValue>(Func<TValue, TNewValue> valueConver
                 .WithReasons(Reasons);
         }
 
+        /// <summary>
+        /// Convert result with value to result with another value that may fail.
+        /// </summary>
+        /// <example>
+        /// <code>
+        ///  var bakeryDtoResult = result
+        ///     .Bind(GetWhichMayFail)
+        ///     .Bind(ProcessWhichMayFail)
+        ///     .Bind(FormattingWhichMayFail);
+        /// </code>
+        /// </example>
+        /// <param name="valueConverter">Transformation that may fail.</param>
+        public Result<TNewValue> Bind<TNewValue>(Func<TValue, Result<TNewValue>> valueConverter)
```

Comment:
I'am not really happy with the name valueConverter. Do you have a better name to this?

Reply:
`bind` or `f`, other languages and language-ext call it simply `f`, because the operation is too abstract and the name of the term is meaningless, its type is the storyteller. 
I will go for `bind`.

Reply:
Renamed